libraries/chain/include/graphene/chain/protocol/offer.hpp

```diff
@@ -0,0 +1,148 @@
+         //  * minimum_price.asset_id == maximum_price.asset_id.
+         //  * to set fixed price without auction minimum_price == maximum_price
+         //  * If buying_item is true, and minimum_price != maximum_price, the user is
+         //  proposing a “reverse auction”
+         //  * where bidders can offer to sell the item for progressively lower prices.
+         //  * In this case, minimum_price functions as the sell-it-now price for the
+         //  reverse auction
+         //  */
+         account_id_type issuer;
+         /// minimum_price is minimum bid price. 0 if no minimum_price required
+         asset minimum_price;
+         /// buy_it_now price. 0 if no maximum price
+         asset maximum_price;
+         /// true means user wants to buy item, false mean user is selling item
+         bool buying_item;
+         /// not transaction expiration date
+         fc::time_point_sec offer_expiration_date;
+
+         /// User provided data encrypted to the memo key of the "to" account
+         optional<memo_data> memo;
+         extensions_type extensions;
+
+         account_id_type fee_payer() const { return issuer; }
+         void validate() const;
+         share_type calculate_fee(const fee_parameters_type &k) const;
+      };
+
+      struct bid_operation : public base_operation
+      {
+         struct fee_parameters_type
+         {
+            uint64_t fee = 20 * GRAPHENE_BLOCKCHAIN_PRECISION;
+            uint32_t price_per_kbyte =
+                10 * GRAPHENE_BLOCKCHAIN_PRECISION; /// only required for large memos.
+         };
+
+         asset fee;
+         account_id_type bidder;
+
+         asset bid_price;
+         offer_id_type offer_id;
+
+         extensions_type extensions;
+
+         account_id_type fee_payer() const { return bidder; }
+         void validate() const;
+         share_type calculate_fee(const fee_parameters_type &k) const;
+      };
+
+      struct cancel_offer_operation : public base_operation
+      {
+         struct fee_parameters_type
+         {
+            uint64_t fee = 20 * GRAPHENE_BLOCKCHAIN_PRECISION;
+            uint32_t price_per_kbyte =
+                10 * GRAPHENE_BLOCKCHAIN_PRECISION;
+         };
+
+         asset fee;
+
+         account_id_type issuer;
+         offer_id_type offer_id;
+
+         extensions_type extensions;
+
+         account_id_type fee_payer() const { return issuer; }
+         void validate() const;
+         share_type calculate_fee(const fee_parameters_type &k) const;
+      };
+
+      enum class result_type
+      {
+         Expired = 0,
+         ExpiredNoBid = 1,
+         Cancelled = 2
+      };
+
+      struct finalize_offer_operation : public base_operation
+      {
+         struct fee_parameters_type
+         {
+            uint64_t fee = 0;
+         };
+
+         asset fee;
+         account_id_type fee_paying_account;
+
+         offer_id_type offer_id;
+
+         fc::enum_type<uint8_t, result_type> result;
```

Comment:
Aren't we changing this to the a normal enum instead of using uint8_t?

Reply:
done